src/common/util-shared.js

```diff
@@ -0,0 +1,31 @@
+"use strict";
+
+const util = require("./util");
+
+module.exports = function(options) {
```

Comment:
Could we delete `locStart` and `locEnd` here, then remove the factory/closure and pass `options` explicitly to functions that need it?

Reply:
Good idea - that also allows more reuse of the already existing `options` argument in the printer. Removing the closure would mean that we also won't pass `util` down to the printer anymore, but that might be a good idea? It's certainly removing an extra abstraction layer ...